support/build/Makefile.rules

```diff
@@ -252,30 +252,29 @@ verbose_cmd = @printf '  %-7s %s\n' '$(1)'  '$(2)' && $(3)
 verbose_cmd_inner = printf '  %-7s %s\n' '$(1)'  '$(2)' && $(3)
 endif
 
-# Writes arguments into a build_args file
+# writes to a file during deferred expansion
 #
-# write_build_args $target,$longargs(optional)
-ifeq ($(CONFIG_BUILD_ARGS_FILE),y)
-write_build_args = $(if $(2),$(file > $(addsuffix .build_args,$(1)),$(2))@$(addsuffix .build_args,$(1)),)
-else
-write_build_args = $(2)
-endif
+# write_deferred $filepath,$content
+write_deferred = $(if $(UK_DEFERRED_EXPANSION),$(file > $(1),$(2)),$$(file > $(1),$(2)))
```

Comment:
Just a minor thing: I‘d call it „file_deferred“. What do you think?

Reply:
The reason why this is called `write_deferred` is that the `file` function can also used to read from a file.  How about `write_file_deferred`?

I don't have a strong opinion on that though, so `file_deferred` would be okay for me.